src/utils/self_test/self_test.c

```diff
@@ -310,33 +441,119 @@ static void print_fail_counts(struct st_latency *latencies,
 
 		local_rep++;
 	}
+}
+
+/*Returns the number of valid points */
+static int calculate_stats(struct st_latency *latencies, int count,
+			   int64_t *av, double *sd, int64_t *min,
+			   int64_t *max, int64_t *med25, int64_t *med50,
+			   int64_t *med75, int64_t *total)
+{
+	uint32_t	i;
+	uint32_t	num_failed = 0;
+	uint32_t	num_passed = 0;
+	double		latency_std_dev = 0;
+	int64_t		latency_avg = 0;
+	int64_t		value;
+	int64_t		lmin = 0;
+	int64_t		lmax = 0;
+	int		idx;
+
+	/* Find initial value for max and min */
+	for (i = 0; i < count; i++) {
+		if (latencies[i].cci_rc == 0) {
+			lmax = latencies[i].val;
+			lmin = latencies[i].val;
+			break;
+		}
+	}
 
+	/* Sum total for average.  Find max/min */
+	for (i = 0; i < count; i++) {
+		if (latencies[i].cci_rc < 0) {
+			num_failed++;
+			continue;
+		}
+		num_passed += 1;
+		value = latencies[i].val;
+		latency_avg += value;
+		lmin = ((lmin < value) ? lmin : value);
+		lmax = ((lmax > value) ? lmax : value);
+	}
+	*total = latency_avg;
+	latency_avg /= num_passed;
+	*av = latency_avg;
+	*min = lmin;
+	*max = lmax;
+
+	/* Find sum square from average (variance) and standard deviation */
+	for (i = 0; i < count; i++) {
+		if (latencies[i].cci_rc < 0) {
+			/* avoid checkpatch warning */
+			continue;
+		}
+		value = latencies[i].val - latency_avg;
+		latency_std_dev += value * value;
+	}
+	latency_std_dev /= num_passed;
+	latency_std_dev =  sqrt(latency_std_dev);
+	*sd = latency_std_dev;
+
+	/* Find medium values.  Works for sorted input only. */
```

Comment:
s/medium/median?

Reply:
Both means the middle position, but median is more restrictive and used for value.  Changed to median.